src/transformers/generation/utils.py

```diff
@@ -1302,7 +1302,10 @@ def _get_initial_cache_position(self, input_ids, model_kwargs):
             if isinstance(model_kwargs["past_key_values"], Cache):
                 past_length = model_kwargs["past_key_values"].get_seq_length()
             else:
-                past_length = model_kwargs["past_key_values"][0][0].shape[2]
+                if isinstance(model_kwargs["past_key_values"][0][0], list):
+                    past_length = sum(x.shape[-2] for x in model_kwargs["past_key_values"][0][0])
```

Comment:
if the past key values is a EfficientCache class, this should be easier to extract. + @gante using the cache_positions would be the best here!